src/rng.jl

```diff
@@ -59,6 +59,15 @@ function Random.seed!(rng::TracedRNG, key)
     return Random123.set_counter!(rng.rng, 0)
 end
 
+"""
+    gen_seeds(trng::TracedRNG, other::Random.AbstractRNG)
+
+Seed a TracedRNG from another 
+"""
+function gen_seeds(::Type{T}, other::Random.AbstractRNG, n::Integer=1) where {T}
+    return Tuple(rand(other, T, n))
```

Comment:
This will allocate and be type unstable. I guess it might be easier to just create a sampler in the function above and then loop through the different rngs, without sampling everything upfront. The sampler is optimized for computing a stream, and not only a single, random number: https://docs.julialang.org/en/v1/stdlib/Random/#Random.Sampler

Reply:
Not quite sure I understand why we need to define the sampler, can we just do define the `SamplerType` ?
```julia
for T = (:UInt32, :UInt64)
    for N in (:1, :2, :4)
        @eval begin
            Base.rand(rng::Random.AbstractRNG, ::Random.SamplerType{NTuple{$N, $T}}) = Tuple(rand(rng, $T, $N))
        end
    end
end 
```

Reply:
This would be a quite severe type piracy...

We don't have to construct a sampler but usually it will be more efficient than calling `rand(rng, UInt32)` etc. in a loop. My point was that I don't think you have to sample a tuple at all but can generate random seeds (at least) as efficiently by using something like
```julia
sampler = Random.Sampler(rng, UInt32)
for subrng in subrngs
    Random.seed!(subrng, rand(rng, sampler))
end
```

Reply:
That makes sense, the only issue is some of these `Sampler` are not defined: the seed for a `Random123.Philox4x` is a `Tuple{UInt64, UInt64}`.

Reply:
In this case you could still use a `UInt64` sampler but sample twice for each `subrng`. I guess you could dispatch on the type of subrng to simplify the code.